### PR TITLE
Modify internal/docker-push interpolation order

### LINKED ISSUE #540: Ensure that internal/docker-push interpolates env vars BEFORE splitting tags
# Current Behavior

When we have following steps:

```
steps:

  - script:
    code: export TAGS_TO_PUSH=tag1, tag2, tag3

  - internal/docker-push:
    tag: latest, $TAGS_TO_PUSH

```

`validateTags` occures an error like

> Invalid tag: tag1,tag2, tag3: invalid tag format

## Cause

- Interpolation is processed AFTER splitting tags.
- https://github.com/wercker/wercker/blob/1.0.1421/docker/docker.go#L468

# Expected Behavior

- Interpolation is processed BEFORE splitting tags.

### PR DESCRIPTION
**Pull request by yuki-takei**
This PR fixes #540 .

# Summary

See [Files Changed](https://github.com/wercker/wercker/pull/541/files?utf8=%E2%9C%93&diff=split).

### CODE REVIEW

File: docker/docker.go [hunk 1]
```diff
@@ -462,12 +462,9 @@ func (s *DockerPushStep) configure(env *util.Environment) error {
 	}
 
 	if tags, ok := s.data["tag"]; ok {
-		splitTags := util.SplitSpaceOrComma(tags)
```
Comment (yuki-takei) [inline docker/docker.go]: e.g.
When `tags := "latest, $TAGS_TO_PUSH"`,

The array like `["latest", "$TAGS_TO_PUSH"]` will be created.

File: docker/docker.go [hunk 2]
```diff
@@ -462,12 +462,9 @@ func (s *DockerPushStep) configure(env *util.Environment) error {
...
-		interpolatedTags := make([]string, len(splitTags))
-		for i, tag := range splitTags {
-			interpolatedTags[i] = env.Interpolate(tag)
-		}
```
Comment (yuki-takei) [inline docker/docker.go]: The array like `["latest", "tag1, tag2, tag3"]` will be created
and the second item is invalid as docker tag.

File: docker/docker.go [hunk 3]
```diff
@@ -462,12 +462,9 @@ func (s *DockerPushStep) configure(env *util.Environment) error {
...
-		s.tags = interpolatedTags
+		interpolatedTags := env.Interpolate(tags)
```
Comment (yuki-takei) [inline docker/docker.go]: The string like "latest, tag1, tag2, tag3" will be created.

File: docker/docker.go [hunk 4]
```diff
@@ -462,12 +462,9 @@ func (s *DockerPushStep) configure(env *util.Environment) error {
...
+		splitTags := util.SplitSpaceOrComma(interpolatedTags)
```
Comment (yuki-takei) [inline docker/docker.go]: The array like `["latest", "tag1", "tag2", "tag3"]` will be created
and these are all valid.

**Comment by markxnelson** [comment]
Please take a look at our CONTRIBUTING.md for information about requirements. Thank you for your contribution

**Comment by yuki-takei** [comment]
@markxnelson 
I have not signed OCA ever, so it looks a little bit pain...

I will be satisfied if this small changes is incorporated anyway. Would you please do that? Or, should I sign to OCA and re-send another PR?

**Comment by markxnelson** [comment]
We cannot merge your PR without you signing the OCA, but we can investigate and fix the issue ourselves.

**Comment by nigeldeakin** [comment]
We've fixed this in PR https://github.com/wercker/wercker/pull/546 so this PR can be closed.

**Pull request closed without merging**